pkg/executables/clusterctl.go

```diff
@@ -346,3 +340,51 @@ func (c *Clusterctl) Upgrade(ctx context.Context, managementCluster *types.Clust
 
 	return nil
 }
+
+func (c *Clusterctl) InstallEtcdadmProviders(ctx context.Context, clusterSpec *cluster.Spec, cluster *types.Cluster, infraProvider providers.Provider, installProviders []string) error {
+	if cluster == nil {
+		return fmt.Errorf("invalid cluster (nil)")
+	}
+	if cluster.Name == "" {
+		return fmt.Errorf("invalid cluster name '%s'", cluster.Name)
+	}
+	clusterctlConfig, err := c.buildConfig(clusterSpec, cluster.Name, infraProvider)
+	if err != nil {
+		return err
+	}
+
+	params := []string{
+		"init",
+		"--config", clusterctlConfig.configFile,
+	}
+
+	for _, provider := range installProviders {
+		switch provider {
+		case constants.EtcdAdmBootstrapProviderName:
+			params = append(params, "--bootstrap", clusterctlConfig.etcdadmBootstrapVersion)
+		case constants.EtcdadmControllerProviderName:
+			params = append(params, "--bootstrap", clusterctlConfig.etcdadmControllerVersion)
+		}
+	}
+
+	// Not supported for docker controllers at this time
+	if clusterSpec.Spec.DatacenterRef.Kind != anywherev1alpha1.DockerDatacenterKind {
+		params = append(params, "--watching-namespace", constants.EksaSystemNamespace)
```

Comment:
I'm lost here, could you explain a bit more why this is not supported for docker?

Reply:
good question, I saw this in the main clusterctl init method in this file above, didn't get a chance to check why it's not supported but thought I should add it in here